src/components/FeatureViews/Registration/RegistrationActions.js

```diff
@@ -37,44 +34,56 @@ query GetRegisteringParent($userId: ID!) {
 }
 `
 
-const RegistrationActions = ({ updateRegisteringParent }) => {
-	const AuthUser = useSelector(({ auth }) => auth);
-	const { parentIsLoggedIn } = useValidateRegisteringParent();
-	const { currentParent, ...registrationState } = getRegistrationCart();
-	const [updatedRegistrationCartNum, setUpdatedRegistrationCartNum] = useState(null);
+const RegistrationActions = () => {
+	const AuthUser = useSelector(({auth}) => auth);
+	const {currentParent, ...registrationState} = useSelector((state) => state.Registration);
+	const {parentIsLoggedIn} = useValidateRegisteringParent();
+	const dispatch = useDispatch();
+
 	const [dialogOpen, setDialog] = useState(false);
-	const { data, error, loading } = useQuery(GET_PARENT_QUERY, {
-		variables: { userId: AuthUser.user.id },
+	const {data, error, loading} = useQuery(GET_PARENT_QUERY, {
+		variables: {userId: AuthUser.user.id},
 		skip: AuthUser.accountType !== USER_TYPES.parent,
 	});
 	const history = useHistory();
 
+	useEffect(() => {
+		dispatch({
+			type: types.INIT_COURSE_REGISTRATION,
+			payload: {},
+		})
+	}, [types.INIT_COURSE_REGISTRATION, dispatch])
+
 	const openDialog = useCallback(() => {
 		setDialog(true);
 	}, []);
 
-	const closeDialog = useCallback((isParentSet) => {
+	const closeDialog = useCallback(() => {
 		setDialog(false);
-		updateRegisteringParent(isParentSet);
 	}, []);
 
 	useEffect(() => {
 		if (parentIsLoggedIn && !loading && Object.values(registrationState).length === 0) {
-			setParentRegistrationCart(data.parent);
-			updateRegisteringParent(false);
+			dispatch({
+				type: types.SET_PARENT,
+				payload: data.parent,
+			})
 		}
 	}, [AuthUser.accountType, loading]);
 
-	if (loading) return <Loading />;
+	if (loading) return <Loading/>;
 	if (error) return <div>There has been an error: {error.message}</div>
 
 	const registeringParent = data?.parent || currentParent;
 
 	const parentName = registeringParent && fullName(registeringParent.user);
-
-	const numberOfRegistrationsInCart = updatedRegistrationCartNum || Object.values(registrationState).reduce((accumulator, currentStudent) => {
-		return accumulator + currentStudent.length;
-	}, 0);
+	const {
+		submitStatus = {},
+		...registrationCartState
+	} = registrationState;
+	const numberOfRegistrationsInCart = Object.values(registrationCartState).length > 0 ?
+		Object.values(registrationState).reduce((accumulator, currentStudent) => (accumulator + currentStudent?.length),
+			0) : 0;
```

Comment:
Ternary not needed, this will simplify to 0 if registrationState is empty anyways